### PR TITLE
Add system stop pulling replication log

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added `SYSTEM STOP/START PULLING REPLICATION LOG` queries (for testing `ReplicatedMergeTree`)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit 5b89a01f52030eaecad466abdf617672b5ac23a0 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52881/5b89a01f52030eaecad466abdf617672b5ac23a0/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

**Comment by tavplubix** [comment]
Initially, I tried to write a test for a race condition between `StorageReplicatedMergeTree::updateQuorum` and `StorageReplicatedMergeTree::cleanLastPartNode`: https://gist.github.com/tavplubix/b2c5d98c0995ff9e986ca6f95827d190

However, this race condition is quite rare and not harmful, nobody cares about it, and a fix would require a backward incompatible change... So it's easier to keep it as is.

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -197,6 +197,7 @@ namespace ActionLocks
     extern const StorageActionBlockType ReplicationQueue;
     extern const StorageActionBlockType PartsTTLMerge;
     extern const StorageActionBlockType PartsMove;
+    extern StorageActionBlockType PullReplicationLog;
```
Comment (evillique) [inline src/Storages/StorageReplicatedMergeTree.cpp:200]:
```suggestion
    extern const StorageActionBlockType PullReplicationLog;
```

**Comment by tavplubix** [comment]
I hate tests for RBAC

**Review by evillique — APPROVED** [review]

**Comment by tavplubix** [comment]
Stress test (asan) - https://github.com/ClickHouse/ClickHouse/issues/52978
Upgrade check is broken in master

**Pull request merged**